docs/marketplace/dev/feature-integration-guides/202108.0/marketplace-product-offer-feature-integration.md

```diff
@@ -1469,6 +1498,65 @@ class ShopApplicationDependencyProvider extends SprykerShopApplicationDependency
 }
 ```
 
+**src/Pyz/Yves/ProductDetailPage/Controller/ProductController.php**
+
+```php
+<?php
+
+namespace Pyz\Yves\ProductDetailPage\Controller;
+
+use Generated\Shared\Transfer\ProductStorageCriteriaTransfer;
+use SprykerShop\Yves\ProductDetailPage\Controller\ProductController as SprykerShopProductController;
+
+class ProductController extends SprykerShopProductController
+{
+    protected function executeDetailAction(array $productData, Request $request): array
+    {
+        $shopContextTransfer = $this->getFactory()
+            ->createShopContextResolver()
+            ->resolve();
+
+        $productStorageCriteriaTransfer = (new ProductStorageCriteriaTransfer())
+            ->fromArray($shopContextTransfer->toArray());
+
+        $productViewTransfer = $this->getFactory()
+            ->getProductStorageClient()
+            ->mapProductStorageData($productData, $this->getLocale(), $this->getSelectedAttributes($request));
+            ->mapProductStorageData($productData, $this->getLocale(), $this->getSelectedAttributes($request), $productStorageCriteriaTransfer);
+        ...
+    }
+}
+```
+
+**src/Pyz/Yves/ProductDetailPage/Theme/default/components/molecules/product-configurator/product-configurator.twig**
+
+```twig
+{%- raw -%}
+...
+{% widget 'AddToCartFormWidget' args [config, data.product, isDisabled, options] only %}
```

Comment:
We do not add such things to the integration guides, project-level twig can be absolutely different from what we have in core. Please get rid of this example.